io/src/read_detector.cpp

```diff
@@ -20,7 +20,8 @@
 namespace {
 
 /// Common implementation for constructing a detector from a set of input files
-template <typename detector_t>
+/// @tparam CAP Grid bin capacity 0: dynamic bin capacity
+template <typename detector_t, unsigned int CAP = 0u>
```

Comment:
I expected that the PR would already be making use of this. 🤔

Unfortunately this is a pretty inconvenient API. 😦 Normally you don't need to explicitly define the `detector_t` type, as the compiler can deduce it from your call. But now, if you want to override the second value, you need to define `detector_t` explicitly as well. 😦

Not to mention that this is sitting in the `.cpp` file. Users will not be able to call this thing anyway.

So in this PR, it would be better to not introduce a new template parameter to this helper function, but just to hardcode the `0u` value on line 41. 🤔